cl/corpus_importer/management/commands/customer_import_project.py

```diff
@@ -0,0 +1,246 @@
+                f"Found a PK in cache, starting import process from record {last_pk}."
+            )
+
+        total_number_of_records = int(
+            r.hget(f"{record_type}_import_status", "total_records") or 0
+        )
+        if not total_number_of_records:
+            total_number_of_records = get_total_number_of_records(
+                record_type, options["use_replica"]
+            )
+            r.hset(
+                f"{record_type}_import_status",
+                "total_records",
+                total_number_of_records,
+            )
+
+        counter = int(
+            r.hget(f"{record_type}_import_status", "next_iteration_counter")
+            or 0
+        )
+        while True:
+            query, params = get_custom_query(
+                options["record_type"],
+                last_pk,
+            )
+            params.append(options["query_batch_size"])
+
+            with connections[
+                "replica" if options["use_replica"] else "default"
+            ].cursor() as cursor:
+                cursor.execute(query, params)
+                rows = cursor.fetchall()
+                record_count = cursor.rowcount
+
+            if not record_count:
+                logger.info("Finished all the records!")
+                break
+
+            # Write the content of the csv file to a buffer and upload it
+            with io.StringIO() as csvfile:
+                writer = csv.DictWriter(
+                    csvfile,
+                    fieldnames=["bucket", "file_name"],
+                    extrasaction="ignore",
+                )
+                for row in batched(rows, options["lambda_record_size"]):
+                    query_dict = {
+                        "bucket": bucket_name,
+                        "file_name": (
+                            f"{row[0][0]}_{row[-1][0]}"
+                            if len(row) > 1
+                            else f"{row[0][0]}"
+                        ),
+                    }
+                    writer.writerow(query_dict)
+
+                s3_client.put_object(
+                    Key=f"{record_type}_filelist_{counter}.csv",
+                    Bucket=bucket_name,
+                    Body=csvfile.getvalue().encode("utf-8"),
+                )
+
+            counter += 1
+            last_pk = rows[-1][0]
+            records_processed = int(
+                r.hget(f"{record_type}_import_status", "records_processed")
+                or 0
+            )
+
+            # Store the results of this iteration and log the progress
+            r.hset(f"{record_type}_import_status", "last_pk", last_pk)
+            r.hset(
+                f"{record_type}_import_status",
+                "next_iteration_counter",
+                counter,
+            )
+            r.hset(
+                f"{record_type}_import_status",
+                "records_processed",
+                records_processed + record_count,
+            )
```

Comment:
You can do this in a single command:

```suggestion
            r.hset(
                f"{record_type}_import_status", 
                mapping={
                    "last_pk": last_pk,
                    "next_iteration_counter": counter,
                    "records_processed": records_processed + record_count,
                },
            )
```

Reply:
Thanks for the suggestion. I'll update the code